config/api/routes/files.php

```diff
@@ -47,17 +51,28 @@
 			// move_uploaded_file() not working with unit test
 			// @codeCoverageIgnoreStart
 			return $this->upload(function ($source, $filename) use ($path) {
-				$props = [
+				// handle chunked uploads
+				$source = Upload::chunk($this, $source, $filename);
+
+				// if file is not yet complete, return
+				if ($source === null) {
+					return;
+				}
+
+				// move the source file from the temp dir
+				$file = $this->parent($path)->createFile([
 					'content' => [
 						'sort' => $this->requestBody('sort')
 					],
 					'source'   => $source,
 					'template' => $this->requestBody('template'),
 					'filename' => $filename
-				];
+				], true);
 
-				// move the source file from the temp dir
-				return $this->parent($path)->createFile($props, true);
+				// clean up tmp
+				F::remove($source);
```

Comment:
As the file is moved above, is there a case where the source file still exists at this point?

Reply:
Need to double check, but I think in case e.g. FileRules refused it. As the move/copy part only takes place after running the rules.